File: Patterns/7. Pyramid Patterns/63. Pyramid Increasing Numbers/pin.py
n = int(input("Enter a Number : "))

for i in range(1,n+1):
    # print spaces
    for j in range(1,n-i+1):
        print(end='  ')
    # print first half
    for j in range(1,i+1):
        print(j,end=' ')
    # print second half
    for j in range(i-1,0,-1):
        print(j,end=' ')
    print()


'''

Output:

        1 
      1 2 1 
    1 2 3 2 1 
  1 2 3 4 3 2 1 
1 2 3 4 5 4 3 2 1 

'''

File: Patterns/07. Pyramid Patterns/63. Pyramid Increasing Numbers/pin.java
import java.util.*;
public class pin{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        for(int i=1; i<=n; i++){
            // print spaces
            for(int j=1; j<=n-i; j++){
                System.out.print("  ");
            }
            // print first half
            for(int j=1; j<=i; j++){
                System.out.print(j + " ");
            }
            // print second half
            for(int j=i-1; j>=1; j--){
                System.out.print(j + " ");
            }
            System.out.println();
        }
    }
}

/*
Output

        1 
      1 2 1 
    1 2 3 2 1 
  1 2 3 4 3 2 1 
1 2 3 4 5 4 3 2 1 

*/